lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/EvaluatorRuntime.cs

```diff
@@ -241,6 +250,11 @@ private void OnException(Exception e)
             }
         }
 
+        private void UnhandledExceptionHandler(object sender, UnhandledExceptionEventArgs e)
```

Comment:
It also will not work at the Task level due to the dependency on `EvaluatorRuntime`. i.e. The Injector is forked by a child component of `ContextManager` several times prior to injecting the `Task`, but `ContextManager` has a dependency on `IHeartBeatManager`, which in turn has an `IInjectableFuture<EvaluatorRuntime>`. This means that the `EvaluatorRuntime` cannot be instantiated prior to injecting the `Task`, so injecting the `EventHandler` will create another instance of `EvaluatorRuntime`.